server/src/main/java/org/elasticsearch/action/admin/indices/get/GetIndexResponse.java

```diff
@@ -126,6 +131,13 @@ public GetIndexResponse(String[] indices,
             defaultSettingsMapBuilder.put(in.readString(), Settings.readSettingsFromStream(in));
         }
         defaultSettings = defaultSettingsMapBuilder.build();
+
+        ImmutableOpenMap.Builder<String, String> dataStreamsMapBuilder = ImmutableOpenMap.builder();
+        int dataStreamsSize = in.readVInt();
+        for (int i = 0; i < dataStreamsSize; i++) {
+            dataStreamsMapBuilder.put(in.readString(), in.readOptionalString());
+        }
```

Comment:
I think this de-serialization logic needs to be inside a version check.
I think this is why the mixed cluster tests have failed.